### PR TITLE
refactor: add clang style check for samples

### PR DESCRIPTION
**Pull request by generalova-kate**
### Details:
 - Added ngraph clang style checks for IE Samples

### Tickets:
 - 49432

### CODE REVIEW

File: inference-engine/samples/style_transfer_sample/main.cpp [hunk 1]
```diff
@@ -2,48 +2,50 @@
...
-    if (FLAGS_i.empty()) {
-        showUsage();
-        throw std::logic_error("Input is required but not set. Please set -i option.");
-    }
-
-    return true;
+  slog::info << "Parsing input parameters" << slog::endl;
+
+  gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+  if (FLAGS_h) {
```
Comment (ilya-lavrenov) [inline inference-engine/samples/style_transfer_sample/main.cpp:30]: why 2 spaces? Usually we have 4

Comment (generalova-kate) [reply inference-engine/samples/style_transfer_sample/main.cpp:30]: I forgot .clang-format

**Comment by ilyachur** [comment]
@generalova-kate Do you have a plan to update the PR and at least add the `.clang-format` configuration file?

**Comment by generalova-kate** [comment]
> @generalova-kate Do you have a plan to update the PR and at least add the `.clang-format` configuration file?

of course.

File: inference-engine/samples/style_transfer_sample/main.cpp (hunk 1 above)
Comment (generalova-kate) [reply inference-engine/samples/style_transfer_sample/main.cpp]: added

File: inference-engine/.clang-format [hunk 2]
```diff
@@ -0,0 +1,54 @@
+BasedOnStyle: LLVM
```
Comment (ilyachur) [inline inference-engine/.clang-format:1]: As I know at the current moment we use the google style (it looks like this config: https://github.com/openvinotoolkit/openvino/blob/master/thirdparty/itt_collector/.clang-format, I used cpplint rules to create this file).

Am I right that that you have used nGraph config as a baseline?

@ilya-lavrenov What do you think should we align samples with nGraph or IE code style?

In additional, if you enable clang-format only for samples, please add this file only to samples folder, because in other case developers, who have integrated clang-format to editors, will have problems, because editor will automatically apply this code style to all changed files.

File: cmake/developer_package/clang_format/clang_format_check.cmake [hunk 3]
```diff
@@ -7,12 +7,13 @@ file(REMOVE "${OUTPUT_FILE}")
...
-# Display the cpplint output to console (to parse it form IDE)
-message("${output}")
+
 file(WRITE "${OUTPUT_FILE}" "${STYLE_CHECK_RESULT}")
 
 if(NOT SKIP_RETURN_CODE)
     if("${STYLE_CHECK_RESULT}" MATCHES ".*<replacement .*")
-        message(FATAL_ERROR "[clang-format] Code style check failed for : ${INPUT_FILE}")
+        # Display the style check output to console in case an issue
+        message("Style check result: ${STYLE_CHECK_RESULT}")
```
Comment (ilyachur) [inline cmake/developer_package/clang_format/clang_format_check.cmake:16]: Why did you add these changes to cmake config? If I remember right the value of `${STYLE_CHECK_RESULT}` contains a lot of information and it is not user-friendly to read

Comment (generalova-kate) [reply cmake/developer_package/clang_format/clang_format_check.cmake:16]: the previous message wasn't useful to it goes to the nonexistence file. I added it to get a little bit of info about clang-format issues.

File: inference-engine/.clang-format (hunk 2 above)
Comment (generalova-kate) [reply inference-engine/.clang-format:1]: I take this format file from https://github.com/openvinotoolkit/openvino/tree/master/ngraph/ folder. Ok I will move it to the sample folder.

Comment (generalova-kate) [reply inference-engine/.clang-format]: I took google style.

File: cmake/developer_package/clang_format/clang_format_check.cmake (hunk 3 above)
Comment (generalova-kate) [reply cmake/developer_package/clang_format/clang_format_check.cmake]: remove it at all.

File: inference-engine/samples/CMakeLists.txt [hunk 4]
```diff
@@ -248,12 +248,11 @@ macro(ie_add_sample)
     endif()
     add_dependencies(ie_samples ${IE_SAMPLE_NAME})
 
-    if(COMMAND add_cpplint_target AND NOT IE_SAMPLE_EXCLUDE_CPPLINT)
-        if(c_sample)
-            set(custom_filters "-readability/casting")
+    if(COMMAND add_clang_format_target AND NOT IE_SAMPLE_EXCLUDE_CLANG_FORMAT)
```
Comment (ilya-lavrenov) [inline inference-engine/samples/CMakeLists.txt:251]: we need to add `EXCLUDE_CLANG_FORMAT` to `set(options EXCLUDE_CPPLINT)`

File: inference-engine/samples/CMakeLists.txt [hunk 5]
```diff
@@ -248,12 +248,11 @@ macro(ie_add_sample)
...
+        add_clang_format_target(${IE_SAMPLE_NAME}_clang FOR_SOURCES ${IE_SAMPLE_SOURCES})
+        if(IE_SAMPLE_HEADERS)
+            add_clang_format_target(${IE_SAMPLE_NAME}_headers_clang FOR_SOURCES ${IE_SAMPLE_HEADERS})
         endif()
```
Comment (ilya-lavrenov) [inline inference-engine/samples/CMakeLists.txt:255]: can we merge these 2 `add_clang_format_target` into a single one? We can check both headers and source files using a single command.

File: inference-engine/samples/common/format_reader/CMakeLists.txt [hunk 6]
```diff
@@ -41,3 +41,7 @@ target_include_directories(${TARGET_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}"
 
 set_target_properties(${TARGET_NAME} PROPERTIES COMPILE_PDB_NAME ${TARGET_NAME}
                                                 FOLDER cpp_samples)
+
+if(COMMAND add_clang_format_target AND NOT IE_SAMPLE_EXCLUDE_CLANG_FORMAT)
```
Comment (ilya-lavrenov) [inline inference-engine/samples/common/format_reader/CMakeLists.txt:45]: `IE_SAMPLE_EXCLUDE_CLANG_FORMAT` is not available here. I think we can drop this check

File: inference-engine/samples/common/utils/CMakeLists.txt [hunk 7]
```diff
@@ -17,3 +17,7 @@ target_link_libraries(${TARGET_NAME}
     PUBLIC
         IE::inference_engine
         gflags)
+
+if(COMMAND add_clang_format_target AND NOT IE_SAMPLE_EXCLUDE_CLANG_FORMAT)
```
Comment (ilya-lavrenov) [inline inference-engine/samples/common/utils/CMakeLists.txt:21]: drop `IE_SAMPLE_EXCLUDE_CLANG_FORMAT` check

File: inference-engine/samples/CMakeLists.txt (hunk 5 above)
Comment (generalova-kate) [reply inference-engine/samples/CMakeLists.txt:255]: I tried but I didn't find to do it.

File: inference-engine/ie_bridges/c/samples/.clang-format [hunk 8]
```diff
@@ -0,0 +1,25 @@
+BasedOnStyle: Google
```
Comment (ilya-lavrenov) [inline inference-engine/ie_bridges/c/samples/.clang-format:1]: I suppose these files will be installed as a part of these commands https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/CMakeLists.txt#L75-L112

can we add the to exclude list?

Secondly, should not we have the common clang-format config for the whole IE? CC @ilyachur are going to align IE and ngraph code styles?

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: inference-engine/ie_bridges/c/samples/.clang-format (hunk 8 above)
Comment (ilyachur) [reply inference-engine/ie_bridges/c/samples/.clang-format:1]: We shouldn't add the global clang-format config, because in this case clang-format will be automatically applied for all changed files if developer integrated clang-format to text editor.
